spec/services/save_provider_user_spec.rb

```diff
@@ -0,0 +1,43 @@
+require 'rails_helper'
+
+RSpec.describe SaveProviderUser do
+  let(:provider) { create(:provider) }
+  let(:new_provider_user_from_form) {
+    SupportInterface::ProviderUserForm.new(
+      email_address: 'test+invite_provider_user@example.com',
+      first_name: 'Firstname',
+      last_name: 'Lastname',
+      provider_ids: [provider.id],
+    ).build
+  }
+  let(:permissions) { { manage_users: [provider.id] } }
+
+  describe '#initialize' do
+    it 'requires a provider_user:' do
+      expect { described_class.new }.to raise_error(ArgumentError)
+      expect { described_class.new(provider_user: ProviderUser.new) }.not_to raise_error
+    end
+  end
+
+  describe '#call!' do
+    subject(:service) do
+      described_class.new(provider_user: new_provider_user_from_form, permissions: permissions)
+    end
+
+    it 'saves the provider user' do
+      expect { service.call! }.to change(ProviderUser, :count).by(1)
+    end
+
+    it 'updates permissions for the saved user' do
```

Comment:
Is it worth adding a spec for the `sanitize_permissions` behaviour here? ie if I try to assign a permission for a provider I don't belong to, it shouldn't work

Reply:
:+1: Thanks @duncanjbrown adding this to the spec helped me spot some unintentional behaviour where an invalid provider id would stop any permissions being set for the given key.
I've changed the method to filter out unassigned provider ids and retain valid ones.